### ISSUE TITLE
Package for MacOSX

### ISSUE THREAD

**Issue by jacob3141** [qt-pods/qt-pods#11]
Compile binaries and package for MacOSX.

**Comment by sgaist** [comment]
Hi,

Need a hand with that part ?

**Comment by jacob3141** [comment]
You're welcome! Feel free to add issues, too, if you encounter bugs :)

**Comment by jacob3141** [comment]
Hey Samuel,

steht das Angebot noch? Ich strebe den ersten Release für Ende März an.

VGJacob

Am 7. Februar 2015, hat Samuel Gaist notifications@github.com geschrieben:

**Comment by sgaist** [comment]
Hey Jacob,

Yes it does

**Comment by jacob3141** [comment]
Done.

**Comment by sgaist** [comment]
You did the package yourself ?

**Comment by jacob3141** [comment]
Yes, you can download it here: http://qt-pods.org/downloads/qtpods-macosx-76d1de6-release.zip

It was easier than I thought. I still need help with the deb package though.

**Comment by sgaist** [comment]
Just downloaded and it can't be opened

By the way, why not a dmg ? That's what OS X users expect to get

**Comment by jacob3141** [comment]
Are you on Yosemite?

You're right, should be a dmg.

edit:
Works fine here on Lion 10.7.5, but was packaged and tested on Yosemite. I am just going to upgrade this machine to Yosemite.

**Comment by sgaist** [comment]
Moutain Lion here

I'd say it's worth keeping 10.7.5, it might be handy from time to time as not everybody is migrating to the latest

**Comment by uranusjr** [comment]
Just a suggestion, would it be better to release this as a Homebrew formula? I can help out if you need a hand.

**Comment by jacob3141** [comment]
I am not using MacOSX frequently as I am actively promoting free software, so I really have no clue on that. Having that said, I don't want to exclude anyone using their system of choice and if there is anything I can do to make life of those easier using a proprietary operating system, I'll gladly help out and incorporate things into the project as long as they're compliant with the license.

@uranusjr @samuelgaist I guess you're using MacOSX, what would you say?

**Comment by sgaist** [comment]
That's not a bad idea but then I'd say a macport recipe would also be a good thing to reach a wider audience since both projects goal is to provide open source software for OS X.

**Comment by jacob3141** [comment]
As I said, whatever I am provided with I will happily integrate it.

**Comment by uranusjr** [comment]
I just made a simple Homebrew formula that builds QtPods. Haven’t tested much beyond basic CLI usage, but things seem to work. The formula is available [here](https://github.com/uranusjr/homebrew/blob/qt-pods/Library/Formula/qt-pods.rb). Since QtPods doesn’t have a stable release yet, the formula is currently HEAD-only. To test it:

``` bash
cd $(brew --prefix)
git fetch https://github.com/uranusjr/homebrew.git qt-pods:qt-pods
git checkout qt-pods
brew install --HEAD qt-pods
```

**Comment by jacob3141** [comment]
Really cool, thank you very much!

**Comment by jacob3141** [comment]
@uranusjr I tested it here, but the executable qt-pods-gui is missing (qt-pods-cli works!). Maybe just an additional target to be added?

**Comment by uranusjr** [comment]
@cybercatalyst Homebrew builds GUI applications as bundles, and does not link them by default. You can find the `.app` bundle inside `$(brew --prefix)/Cellar/qt-pods/HEAD/bin`. Homebrew also offers a separate command to link app bundles to `/Applications`:

``` bash
brew linkapps qt-pods
```

I didn’t add any special configuration for this since it is only kind of a proof of concept, and honestly I’m not even sure what those GUI applications are usually handled. I’ll need to take a look at how other Homebrew formulae do this first.
